cmd/evergreen/evergreen.go

```diff
@@ -107,12 +108,11 @@ func buildApp() *cli.App {
 		if err != nil {
 			return errors.Wrap(err, "problem loading configuration")
 		}
-		if conf.AutoUpgradeCli {
+		if conf.AutoUpgradeCLI {
 			ctx, cancel := context.WithCancel(context.Background())
 			defer cancel()
-			err = operations.DoUpdate(conf, ctx, true, false)
-			if err != nil {
-				return errors.Wrap(err, "problem auto updating CLI")
+			if err = operations.CheckAndUpdateVersion(conf, ctx, true, false); err != nil {
```

Comment:
One concern I have is that if this runs before every command, it'll always say "Binary is already up to date - not updating." because of `checkUpdate` at the beginning. This might be undesirable if people have things like scripts that use the CLI and don't want this extra output. Would it be possible to propagate `silent = true` to `checkUpdate` when auto-upgrades are enabled so that people only see the update logs when there's actually a new update available?

Reply:
That makes sense, yes